### PR TITLE
Automated workspace pruning/cleaning

### PR DESCRIPTION
**Pull request by wangpatrick57**
**Summary**: A command to find and remove unneeded files from the workspace directory.

**Demo**:
Passing 25 unit tests with many edge cases which create files/directories, call `clean_workspace()`, and then verify its contents.
Screenshot 2024-07-07 at 17 05 07

**Details**
* "Aggressive" mode removes all task_runs/\*/ directories that are not directly pointed to by a symlink in symlinks/.
* "Safe" mode also keeps task_runs/\*/ directories which are indirectly pointed to by a symlink. This can happen if a symlink points to a task_runs/\*/ directory which has a symlink in it that points to another task_runs/*/ directory.
* I chose to write so many unit tests because this operation must be bug-free.

### CODE REVIEW

File: manage/cli.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        for child_fordpath in dbgym_cfg.dbgym_runs_path.iterdir():
+            if child_fordpath not in task_run_child_fordpaths_to_keep:
+                if child_fordpath.is_dir():
+                    shutil.rmtree(child_fordpath)
+                else:
+                    os.remove(child_fordpath)
+    ending_num_files = _count_files_in_workspace(dbgym_cfg)
+
+    if verbose:
+        print(f"Removed {starting_num_files - ending_num_files} out of {starting_num_files} files")
```
Comment (lmwnshn) [inline manage/cli.py:210]: Nitpick: consider making a logger for the print calls from this file

**Review by lmwnshn — APPROVED** [review]
LGTM
